### ISSUE TITLE
Deleting job raises KeyError

### ISSUE THREAD

**Issue by remram44** [VisTrails/VisTrails#1114]
Pressing the DEL key to remove a job from the "Running jobs" dialog can raise KeyError:
```
Traceback (most recent call last):
  File "vistrails/run.py", line 144, in <module>
    main()
  File "vistrails/run.py", line 138, in main
    v = app.exec_()
--- Exception caught here ---
  File "/home/remram/Documents/programming/dat/vistrails/vistrails/gui/job_monitor.py", line 267, in keyPressEvent
    parent.parent().jobMonitor.deleteJob(item.job.id)
  File "/home/remram/Documents/programming/dat/vistrails/vistrails/core/vistrail/job.py", line 442, in deleteJob
    self.callback().deleteJob(id)
  File "/home/remram/Documents/programming/dat/vistrails/vistrails/gui/job_monitor.py", line 346, in deleteJob
    del workflow_item.jobs[id]
KeyError: u'5cee0ba1a068a06f67e0982b7f5689baea4c3991'
```

**Comment by rexissimus** [comment]
This should be fixed by yesterdays commit 69d8da4114f3a0f3920e6fa75d95146337129527?

**Comment by remram44** [comment]
Looks that way!
